sentry_sdk/tracing.py

```diff
@@ -555,6 +565,112 @@ def to_json(self):
 
         return rv
 
+    def _set_initial_sampling_decision(self, sampling_context):
+        # type: (SamplingContext) -> None
+        """
+        Sets the transaction's sampling decision, according to the following
+        precedence rules:
+
+        1. If a sampling decision is passed to `start_transaction`
+        (`start_transaction(name: "my transaction", sampled: True)`), that
+        decision will be used, regardlesss of anything else
+
+        2. If `traces_sampler` is defined, its decision will be used. It can
+        choose to keep or ignore any parent sampling decision, or use the
+        sampling context data to make its own decision or to choose a sample
+        rate for the transaction.
+
+        3. If `traces_sampler` is not defined, but there's a parent sampling
+        decision, the parent sampling decision will be used.
+
+        4. If `traces_sampler` is not defined and there's no parent sampling
+        decision, `traces_sample_rate` will be used.
+        """
+
+        def _inherit_or_use_static_rate(parent_sampled, traces_sample_rate):
```

Comment:
Put this into the global namespace, otherwise there's a function that's constantly being defined and discarded, on the hot path

Reply:
actually this is only used once, just inline it (max perf!!!!)

Reply:
TBH, it was more about giving a name than anything else, LOL. I can switch that to a comment and inline it.